plugin/src/main/kotlin/nl/avisi/anowalke/service/RestService.kt

```diff
@@ -0,0 +1,52 @@
+package nl.avisi.anowalke.service
+
+import com.google.gson.Gson
+import com.intellij.openapi.diagnostic.Logger
+import nl.avisi.anowalke.dto.ExpressionsDto
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.io.IOException
+
+class RestService {
+
+    private val apiUrl = "http://localhost:8080"
+
+    private val log = Logger.getInstance(this::class.java.name)
+
+    // Instantiates the OkHttpClient.
+    var client: OkHttpClient = OkHttpClient()
+
+    val gson = Gson()
+    val JSON: MediaType? = ("application/json; charset=utf-8").toMediaTypeOrNull()
+
+
+    @Throws(IOException::class)
+    fun makeApiCall(expressions: ExpressionsDto, url: String): List<String> {
+        val requestBody = gson.toJson(expressions)
+        log.warn(requestBody)
+        val body: RequestBody = requestBody.toRequestBody(JSON)
+        val request: Request = Request.Builder()
+            .url(url).post(body)
+            .addHeader("Content-type", "application/json").build()
```

Comment:
Hier kan je MediaType.APPLICATION_JSON ook gebruiken i.p.v. magic string

Reply:
Hij pakt hier alleen een string. Ik kan er een minder magische string van maken maar daar blijft het bij

Reply:
Je kan kijken of er toch niet nog ergens een enum voor bestaat, waar je uiteindelijk de string waarde van pakt. Huidige oplossing met in een variabele zetten is al beter. Nog beter is om een companion object aan te maken met daarin constanten.